File: src/lib/types/consultation.ts
import type { Appointment } from "./appointment";
import type { User, Users } from "./users";
import type { Doctor } from "./users/doctor";

export interface Consultation {
    id: number;
    doctor: Doctor;
    patient: User<Users.Patient>;
    appointment: Appointment;

    notes: string;
    prescriptions: string[];
    attachments?: string[]; // file URLs

    createdAt: Date;
    updatedAt?: Date;
}

File: frontend/backend/models/consultation.go
package models

import (
	"time"
)

// Consultation represents a medical consultation record
type Consultation struct {
	ID            uint         `gorm:"primaryKey" json:"id"`
	DoctorID      uint         `gorm:"not null;index" json:"doctorId" validate:"required"`
	Doctor        *Doctor      `gorm:"foreignKey:DoctorID" json:"doctor,omitempty"`
	PatientID     uint         `gorm:"not null;index" json:"patientId" validate:"required"`
	Patient       *Patient     `gorm:"foreignKey:PatientID" json:"patient,omitempty"`
	AppointmentID uint         `gorm:"index" json:"appointmentId"`
	Appointment   *Appointment `gorm:"foreignKey:AppointmentID" json:"appointment,omitempty"`
	Notes         string       `gorm:"type:text" json:"notes"`
	Prescriptions string       `gorm:"type:text" json:"prescriptions"` // JSON array
	Attachments   string       `gorm:"type:text" json:"attachments"`   // JSON array of file URLs
	CreatedAt     time.Time    `json:"createdAt"`
	UpdatedAt     time.Time    `json:"updatedAt"`
}

func (Consultation) TableName() string {
	return "consultations"
}
